microsoft/PowerToys
# Change warnings from icon to flyouts but it no longer interrupts searching

## Summary of the Pull Request
This PR reverts the warning system to flyouts since most of the bug bash feedback revolved around warnings not being visible/noticeable. The main issue with the flyout system was that drop downs were almost unusable since searching on them would trigger flyouts even if you dont select them.


## PR Checklist
* [X] Applies to #2778 , #2665


## Detailed Description of the Pull Request / Additional comments
- Removed the warning exclamation icon
- Added the old flyout design for warnings introduced in #2140 
- The main issue with the flyouts was that it made searching in drop downs very difficult. The [ComboBox.SelectionChanged](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.primitives.selector.selectionchanged?view=winrt-18362) handler gets called even when searching in the drop down and a selection is highlighted. This would cause the flyout to appear and it would close the drop down. This caused issues such that if the user cannot type A because it's the first drop down, they could not search for Alt either since A would be the first selection and the flyout would pop up.
- Based on an idea from [this stackoverflow answer](https://stackoverflow.com/questions/16966264/what-event-handler-to-use-for-combobox-item-selected-selected-item-not-necessar), I moved the selection handler to the "Drop Down closed" event. By doing this the selection logic and validation of the shortcut is done only when the drop down is closed, so searching is not affected by this. This however would not work for the following two scenarios:
1. The user uses Type Key/shortcut to enter a key/shortcut and presses OK. In this case the drop downs get populated but it is never opened/closed so that method would not get triggered
2. When opening remap key/shortcuts if you have old remappings then those would have to be reloaded. Again in this case dropdownClosed is not triggered.
3. If the user has keyboard focus on the combo box and just types in something, dropDownClosed isnt triggered.
- To handle the above to cases we have to use the SelectionChangedHandler, however we have to ensure we only do this only for those cases. This is done by checking if the drop down is currently closed. This encompasses all the above cases and the only case where it still happens apart from the above list is if we SetSelectedIndex from within the DropDownClosed handler. Since we only set it to -1 there, and we chave checked for SelectedIndex !=-1 in the selectionchanged logic this is not a problem. All other cases where you have to open the drop down and you click something do not trigger SelectionChanged because SelectionChanged is called first, then the dropDownClosed handler is called, so the drop down is considered open during that.
- After switching to the flyout case there was one bug with using None. If you would set None and the drop down gets deleted but the resulting shortcut has a conflict with another shortcut then an error would occur because a flyout would try to open on that deleted drop down. This has been resolved by doing the drop down deletion only after ensuring that there are no other errors/conflicts if that drop down was removed.


## Validation Steps Performed
Validated all the error messages still appear as before and ensured that warnings appear as they should when using the following:
- Type Shortcut/Key 
- Opening remap keyboard/shortcuts with old remappings and just clicking apply should maintain all the remappings (to ensure selection handlers are correctly called)
- Using the entire UI both using keyboard and using keyboard+mouse and navigating with the drop down and searching using them.
